### PR TITLE
Add roundTo(BucketStart|NextBucket) to UTCInstant

### LINKED ISSUE #232: Separate clean-up and new functionalities in #215
The current PR #215 has a lot of cleanups that include converting `long` to `UTCInstant`. This hides the original issue described in the PR, so I propose to do the same as for #213, which is to split it into two PRs.

I think one could directly apply to `develop`, while the other probably has to depend on #213.

Making a clean-up PR, the only place I see a _real_ new functionality is https://github.com/DP-3T/dp3t-sdk-backend/pull/215/files#diff-f1905691732404c7d82d000947c50bbeR101

Everything else is just cleaning up.

Also, the `getMaxExposedIdForKeyDateDEBUG` and `getSortedExposedForKeyDateDEBUG` are defined, but never used, and only call the methods without the `DEBUG`.

So perhaps I made a wrong new PR? Or is it really just the new sql-expression in `getMaxExposedIdForKeyDate` and `getSortedExposedForKeyDate`?

### PR DESCRIPTION
**Pull request by ineiti**
This PR adds the two methods roundTo(BucketStart|NextBucket) to the 'UTCInstant' class.
This simplifies a lot of calculations in different parts of the code.
More conversions of 'long' to 'UTCInstant'.

Closes #232

### CODE REVIEW

File: dpppt-backend-sdk/dpppt-backend-sdk-model/src/main/java/org/dpppt/backend/sdk/utils/UTCInstant.java [hunk 1]
```diff
@@ -122,6 +122,26 @@ public LocalTime getLocalTime() {
...
+  }
+
+  /**
+   * @param releaseBucketDuration
+   * @return the start of the next bucket of this UTCInstant. If the UTCInstant is at the beginning
+   *     of a bucket, the next bucket will be returned.
+   */
+  public UTCInstant roundToNextBucket(Duration releaseBucketDuration) {
+    var rounding = releaseBucketDuration.toMillis();
+    return new UTCInstant(((this.timestamp / rounding) + 1) * rounding);
```
Comment (ubamrein) [inline dpppt-backend-sdk/dpppt-backend-sdk-model/src/main/java/org/dpppt/backend/sdk/utils/UTCInstant.java:142]: I think this is dangerous for two reasons:

1) it relies on the correct type inference of `rounding` since it is `var`

2) it relies on integer division and no automatic type casts. Justs to give an example: Compare the output of `1/3` in a python2 interpreter and in a python3 (tested with 3.8.5) interpreter. Since they changed the API internally the latter is 0 and the former 0.333. Currently, Java enforces the output type to be the same as the two input type (way of least casts), but I'm not sure if we should relay on it.

So the first error is quite easily introduced when the signature for `toMillis()` changes (or a completely different method is used), and is near to impossible to see without debugging or tooltips.

The second error is just theoretical, since I personally saw this change happen (in python). There is also no way of figuring this error out, since the syntax in both version is correct. Nevertheless, it is safe to assume (I think @martinalig  has some input?) that the Java implementation will not in the near future change this behaviour.

So TL;DR my suggestion is to either use modulo calculus, or at least have a fixed type for rounding.

Comment (martinalig) [reply dpppt-backend-sdk/dpppt-backend-sdk-model/src/main/java/org/dpppt/backend/sdk/utils/UTCInstant.java:142]: I prefer fixed types of var anyway. So lets use the _long_ there. Besides that I do not see any issue here. Also we have some unittests which help us.

**Review by martinalig — APPROVED** [review]

**Review by martinalig — APPROVED** [review]
